### PR TITLE
Routing

### LINKED ISSUE #10: Loading page
# User-Story

As a user

I want to have a loading page

because it looks nicer when i open the app

# Notes, Links, Material
image

# ToDos

- [x] Add title
- [x] Add background color
- [x] possibly add a gif

### LINKED ISSUE #18: React Router
# User-Story

As a developer

I want to install react router 

so that i can add pages and link them

# Notes, Links, Material
https://reactrouter.com/web/guides/quick-start
# ToDos

- [x] Install react router

### LINKED ISSUE #21: Routing
# User-Story

As a developer

I want to add routes

so that users can navigate between pages

# Notes, Links, Material
https://reactrouter.com/web/example/basic

# ToDos

- [x] Link buttons with pages

#18

### PR DESCRIPTION
**Pull request by kschmitz123**
I've added basic routes to navigate between the pages.
Routes to specific projects are hardcoded for now

close #21 
close #18
close #10

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,12 +1,45 @@
...
+
   return (
-    <>
+    <Router>
       <GlobalStyle />
-      <div className="App">Test deployment</div>
-    </>
+      <Switch>
+        <Route exact path="/">
+          {loading === false ? <HomePage /> : <LoadingScreen />}
```
Comment (FabianAlleblas) [inline client/src/App.js]: If loading is a bool you could try `{loading ? <LoadingScreen/> : <HomePage/>}`

File: client/src/App.js [hunk 2]
```diff
@@ -1,12 +1,45 @@
...
+        <Route path="/category">
+          <CategoryPage />
+        </Route>
+        <Route path="/browse">
+          <BrowsePage />
+        </Route>
+        <Route path="/details">
+          <DetailsPage />
+        </Route>
+        <Route path="/favourites">
```
Comment (FabianAlleblas) [inline client/src/App.js:35]: 🇬🇧

File: client/src/components/LoadingScreen.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import styled from "styled-components/macro";
+import Gif from "../assets/loading_animation.gif";
```
Comment (lmachens) [inline client/src/components/LoadingScreen.js]: Consider renaming this import to clarify that you only receive a path/string.

```suggestion
import loadingAnimationSrc from "../assets/loading_animation.gif";
```

File: client/src/pages/HomePage.js [hunk 4]
```diff
@@ -1,33 +1,26 @@
...
+// const SearchButton = styled.button`
+//   background: none;
+//   border: none;
+//   color: var(--button-color);
+//   cursor: pointer;
+//   z-index: 2;
+//   position: fixed;
+//   top: 8px;
+//   right: 10px;
+// `;
```
Comment (lmachens) [inline client/src/pages/HomePage.js]:
```suggestion
```

**Review by lmachens — COMMENTED** [review]
gj 🎉

File: client/src/components/Categories.js [hunk 5]
```diff
@@ -4,8 +4,9 @@ import Sewing from "../assets/sewing-machine.svg";
 import Macrame from "../assets/macrame.svg";
 import Paint from "../assets/color-palette.svg";
 import Woodwork from "../assets/woodworking.svg";
+import { Link } from "react-router-dom";
 
-const CategoryButton = styled.button`
+const NavLink = styled(Link)`
```
Comment (FabianAlleblas) [inline client/src/components/Categories.js:9]: There is a lot positioning going on here, maybe you can simplify that with grid, place-items, place-content or other ways? Hard to vision without a live preview 😄

File: client/src/pages/HomePage.js (hunk 4 above)
Comment (FabianAlleblas) [inline client/src/pages/HomePage.js]: I would remove unused code to keep it clean.

**Review by FabianAlleblas — COMMENTED** [review]
👍  Left a few little comments, nothign major, I hope they help.

File: client/src/App.js (hunk 1 above)
Comment (kschmitz123) [reply client/src/App.js]: thanks for the tip!

File: client/src/components/Categories.js (hunk 5 above)
Comment (kschmitz123) [reply client/src/components/Categories.js:9]: i deployed the app on heroku, link is at the end of my PR :)

Comment (FabianAlleblas) [reply client/src/components/Categories.js:9]: Ok I tried a little bit 😅 

Screenshot 2020-11-26 at 11 17 58

Screenshot 2020-11-26 at 11 18 08

Screenshot 2020-11-26 at 11 18 13

**Pull request merged**